text/0060-data-memos.md

```diff
@@ -0,0 +1,111 @@
+
+This memo will be sent to a recipient of a transaction in the payload `TxOut`. The accompanying change `TxOut` will contain a `DestinationWithDataMemo`.
+
+## Destination With Data Memo
+
+This memo type is similar to the Destination Memo proposed in [RTH](https://github.com/mobilecoinfoundation/mcips/pull/4). Just like the Authenticated Sender Memo in expanded RTH, the Destination memo has 32 bytes of unused space. This memo utilizes that 32 bytes of empty space to store arbitrary data supplied by the client. This memo will have type bytes `0x0204`.
+
+This memo will be written by the sender of a transaction and attached to the change `TxOut`.
+
+## Custom Memo
+
+This memo type offers clients full use of the ememo field (except for the two type bytes). All of the space available for memo data is open for use by the client. No additional data is stored by the memo builder, allowing clients to implement their own memo types.
+
+This memo type has type byes `0x0300` and stores 64 bytes of arbitrary data supplied by the client. It will be written to both the payload and change `TxOut`s of a transaction.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## 0x0103 Authenticated Sender With Data Memo
+
+The 64-byte memo data is laid out as follows:
+
+| Byte range | Item |
+| ---------- | ---- |
+| 0 - 16     | Sender's Address Hash |
+| 16 - 48    | Custom Data           |
+| 48 - 64    | HMAC                  |
+
+This is memo is mostly the same as RTH's authenticated sender memo. Instead of leaving bytes 16-48 unused, however, this space can be filled with aritrary data by the client.
+
+## 0x0204 Destination With Data Memo
+
+The 64-byte memo data is laid out as follows:
+
+| Byte range | Item |
+| ---------- | ---- |
+| 0 - 16     | Recipient's Address Hash |
+| 16 - 17    | The number of recipients, an unsigned 8-bit number |
+| 17 - 24    | Big-endian bytes of fee amount, an unsigned 56-bit number |
+| 24 - 32    | Big-endian bytes of the total outlay amount, an unsigned 64-bit number |
+| 32 - 64    | Custom Data                   |
+
+This is memo is mostly the same as RTH's destination memo. Instead of leaving bytes 32-64 unused, however, this space can be filled with aritrary data by the client.
+
+## 0x0300 Custom Memo
+
+The 64-byte memo data is laid out as follows:
+
+| Byte range | Item |
+| ---------- | ---- |
+| 0 - 64     | Custom Data           |
+
+This memo has no defined structure or data boundaries. All bytes are left for the client to fill with arbitrary data. Developers can create their own memo layouts for their applications using this memo type.
```

Comment:
Do we need to offer this type ? Feel like the above `Authenticated Sender With Data Memo` & `Destination With Data Memo` give flexibility with authenticity guarantees. If clients need more encrypted data they can store a UUID in the 32-bytes offered by above and the rest elsewhere encrypted by the account private keys.

Reply:
I think the sender and destination variants will probably suit most application needs. But I think having the expanded version(s) should all but guarantee developers will have all the tools they need to use memos however they want/need.

I think also think it is important to provide data memos that don't have RTH information. Consider a situation where full anonymity of the sender is required. That can't be achieved if a Sender Memo is written. For instance, suppose Alice is trying to verify the identity of an anonymous person she is transacting with. After receiving the payment, she can compare the address hash with her or Bob's non-anonymous transaction history to look for matches and discover that it came from Charlie. Bob should probably have used a different account for his anonymous activity, but some sort of anonymous payment feature could have saved him.

Reply:
ya you're right the point of this memo type is to give full-control.